### DISCUSSION TITLE
‘-clface’ seems to be abnormal in some cases

### DISCUSSION THREAD

**Discussion by smallrain1996** [neperfepx/neper#333] [Q&A] [answered]
Hello, I use -clface to control the mesh size of the specified surface, but it seems that in some cases the mesh size of some areas has not been adjusted (y==0 everything is normal, but there are some problems with y==10).
Version 4.2.1-14
Thank you very much.
```
neper -T -n 200 -domain "cube(10,10,10)" -statcell size,diameq,group,sphericity -o testy0
neper -M testy0.tess -nset faces -faset none \
-clface "(y==0)?0.3:1" \
-format inp,msh -o testy0
neper -V testy0.msh -print testy0 
```
![testy0](https://user-images.githubusercontent.com/52028117/142728411-29e26dd1-6a91-4df3-babe-efe5759168ad.png)
```
neper -T -n 200 -domain "cube(10,10,10)" -statcell size,diameq,group,sphericity -o testy10
neper -M testy10.tess -nset faces -faset none \
-clface "(y==10)?0.3:1" \
-format inp,msh -o testy10
neper -V testy10.msh -cameracoo 40:60:50 -print testy10
```
![testy10](https://user-images.githubusercontent.com/52028117/142728426-5d079f8d-6749-4246-9eed-f0f0479af069.png)

**Comment by rquey** [comment, marked as answer]
You are dealing with real numbers, so, use a tolerance: `-clface "(y>9.99)?0.3:1"` instead of `-clface "(y==10)?0.3:1"`.
![image](https://user-images.githubusercontent.com/2659376/142828586-7f099f2f-b990-465f-9967-13492496914a.png)

I would also recommend to use `-reg 1` in -T (or at least / instead `-pl 100` in -M) to avoid the local mesh refinements around small edges.
```
neper -T -n 200 -domain "cube(10,10,10)" -statcell size,diameq,group,sphericity -o testy10 -reg 1
neper -M testy10.tess -nset faces -faset none \
-clface "(y>9.99)?0.3:1" \
-format inp,msh -o testy10
neper -V testy10.msh -cameracoo 40:60:50 -print testy10
```
![image](https://user-images.githubusercontent.com/2659376/142828757-e758dc8a-dc59-4219-b691-3671523c7b45.png)

**Reply by smallrain1996** [reply]
Thank you very much for your reply, which not only solved my problem, but also gave me a very important suggestion.
